src/torrent.cpp

```diff
@@ -8646,6 +8646,10 @@ namespace {
 		// to be in downloading state (which it will be set to shortly)
 //		INVARIANT_CHECK;
 
+		// workaround for torrent getting stuck in `downloading` state
+		if (!are_files_checked())
```

Comment:
This change is not intuitive to me. I wonder, if checking torrents always ignored the file- and piece priorities, would that also fix it?